src/tests/string/test_string_functions.f90

```diff
@@ -27,12 +27,21 @@ end subroutine test_to_upper_string
 
     subroutine test_to_title_string
         type(string_type) :: test_string, compare_string
-        test_string = "_#To tiTlE !$%-az09AZ"
-        compare_string = "_#To title !$%-az09az"
+        test_string = "tO t!TL3 7h1S p#ra$e"
+        compare_string = "To T!Tl3 7h1s P#Ra$E"
```

Comment:
Why are the characters "_" and "#" removed?

Reply:
I removed `_#` from the start of the sentence to not give an impression, on someone who is reading the test case, that it is the `_#` which is controlling capitalization of `T`. First alphanum character of the sentence will always be capitalized whether is it is preceded by a non-alphanum character or not. So in short I wanted to improve test case quality because of which I chose to do so, otherwise there is no specific reason for doing so. Please let me know if there are any other ways in which I can further improve this test case.

Reply:
Maybe you could add the test case with "_#". So someone who would read the tests would see the behaviour for both cases.

Reply:
added `_#` in the test case for to_title.